toffy/json_utils_test.py

```diff
@@ -77,3 +78,44 @@ def test_list_moly_fovs(tmpdir):
     pred_moly_fovs = json_utils.list_moly_fovs(tmpdir)
 
     assert np.array_equal(pred_moly_fovs.sort(), moly_fovs.sort())
+
+
+def test_read_json_file():
+
+    with tempfile.TemporaryDirectory() as tmp_dir:
+
+        # create fake jsons
+        moly_json = {'name': 'bob',
+                     'standardTarget': 'Molybdenum Foil'}
+        json_path = tmp_dir+"/test.json"
+        
+        # write test json
+        with open(json_path, 'w') as jp:
+            json.dump(moly_json, jp)
+
+        # Make sure errors come up when the directory is bad. Doesnt check to see
+        # if filename itself exists somewhere else bc that would involve recursion into directories
+        # This relies upon io_utils.validate_paths() being fully functional and tested
+        bad_path = "/neasdf1246ljea/asdfje12ua3421ndsf/asdf.json"
+        with pytest.raises(ValueError, match=f'The path, {bad_path}, is not prefixed with \'../data\'.\n'
+                    f'Be sure to add all images/files/data to the \'data\' folder, '
+                    f'and to reference as \'../data/path_to_data/myfile.tif\''):
+                json_utils.read_json_file(bad_path)
+
+        # Create bad dir path to invoke all 3 possibilities of validate_paths()
```

Comment:
So the testing philosophy is that you don't need to test control flow paths in all helper functions that are called, only the ones that are newly added. So it's not necessary that all cases in `validate_paths` are tested, since that function has its own set of tests. Instead, you just need to test the case where the path is valid, and the case where the path isn't valid